Assets/Scripts/UserInterface/UIButton.cs

```diff
@@ -0,0 +1,70 @@
+/**
+ * Author(s): Ranjini Das
+ **/
+
+using UnityEngine;
+using UnityEngine.UI;
+using System.Collections;
+
+public class UIButton : MannBehavior
+{
+    GameObject gameObject;
+    EventController.Action action; 
+    bool isDown;
+    public delegate void performAction (EventController.Action action);  
+
+    public UIButton (GameObject gameObject, EventController.Action action)
+    {
+        this.gameObject = gameObject;
+        this.action = action; 
+    }
+
+    public void ButtonDown()
+    {
+        isDown = true;
+        AddButtonHeldEvent(action);     
+    }
+
+    public void ButtonUp()
+    {
+        isDown = false;
+        AddOnUpEvent(action);  
+    }
+
+    public bool IsDown()
+    {
+        return isDown; 
+    }
+
+    public void onClick()
+    {
+        AddClickEvent(); 
+    }
+
+    void AddClickEvent(EventController.Action action)
```

Comment:
I think the confusion here is that you're mixing the subscription function and the function to call the pointer events. I'd expect `AddButtonHeldEvent` to subscribe an event whereas `OnButtonHeld` should call the events you've subscribed thus far.